pyproject.toml

```diff
@@ -0,0 +1,104 @@
+[build-system]
+requires = ["setuptools>=61.0", "setuptools_scm[toml]>=6.2"]
+build-backend = "setuptools.build_meta"
+
+[tool.setuptools_scm]
+
+[project]
+name = "pointpats"
+dynamic = ["version"]
+authors = [
+    { name = "Serge Rey", email = "sjsrey@gmail.com" },
+    { name = "Hu Shao", email = "shaohutiger@gmail.com" },
+]
+maintainers = [{ name = "pysal contributors" }]
+license = { text = "BSD 3-Clause" }
+description = "Methods and Functions for planar point pattern analysis"
+keywords = [
+    "spatial statistics, point patterns"
+]
+readme = { text = """\
+"Methods and Functions for planar point pattern analysis"
+
+""", content-type = "text/x-rst" }
+classifiers = [
+    "Programming Language :: Python :: 3",
+    "License :: OSI Approved :: BSD License",
+    "Operating System :: OS Independent",
+    "Intended Audience :: Science/Research",
+    "Topic :: Scientific/Engineering :: GIS",
+]
+requires-python = ">=3.7"
+dependencies = [
+    "numpy",
+    "pandas",
+    "scipy >=1.8",
+    "libpysal >=4",
+    "matplotlib"
+]
+
+[project.urls]
+Home = "https://github.com/pysal/pointpats/"
+Repository = "https://github.com/pysal/pointpats"
+
+[project.optional-dependencies]
+dev = ["pre-commit"]
+docs = [
+    "nbsphinx",
+    "numpydoc",
+    "pandoc",
+    "sphinx",
+    "sphinxcontrib-bibtex",
+    "sphinx_bootstrap_theme",
+    "mkdocs-jupyter",
+    "myst-parser"
+]
+tests = [
+    "geopandas",
+    "codecov",
+    "coverage",
+    "pytest",
+    "pytest-mpl",
+    "pytest-cov",
+    "watermark",
+]
+
+[tool.setuptools.packages.find]
+include = ["pointpats", "pointpats.*"]
+
+[tool.black]
+line-length = 88
+
+[tool.ruff]
+line-length = 88
+select = ["E", "F", "W", "I", "UP", "N", "B", "A", "C4", "SIM", "ARG"]
+target-version = "py39"
```

Comment:
This should technically be `py38`, but seems to be safe to leave as is for now, since we will drop 3.8.